### PR TITLE
Use Azure Storage Accounts as terraform backend

### PR DESCRIPTION
**Pull request by kvmw**

### CODE REVIEW

**Review by maly7 — APPROVED** [review]
Thanks, @kvmw! This looks good. Would you explain what setup and commands you used to create the storage account?

File: terraform/main.tf [hunk 1]
```diff
@@ -50,6 +50,44 @@ resource "azurerm_resource_group" "grp" {
...
+      "Delete",
+      "Purge",
+      "Recover"
+    ]
+  }
+}
+
+# Create Secret for Admin Username
+resource "azurerm_key_vault_secret" "admin_username" {
+  name         = "admin-username"
```
Comment (maly7) [inline terraform/main.tf]: This key vault secret should be named `POSTGRES-LOGIN-NAME` to match what the applications expect as seen [here](https://github.com/Azure-Samples/acme-fitness-store#create-azure-key-vault-and-store-secrets)

File: terraform/main.tf [hunk 2]
```diff
@@ -50,6 +50,44 @@ resource "azurerm_resource_group" "grp" {
...
+  value        = random_password.admin.result
+  key_vault_id = azurerm_key_vault.key_vault.id
+}
+
+# Create Secret for Admin Password
+resource "azurerm_key_vault_secret" "admin_password" {
+  name         = "admin-password"
```
Comment (maly7) [inline terraform/main.tf]: Same thing here, but this should be `POSTGRES-LOGIN-PASSWORD`

**Review by maly7 — CHANGES_REQUESTED** [review]

**Comment by kvmw** [comment]
> Thanks, @kvmw! This looks good. Would you explain what setup and commands you used to create the storage account?

Since this was a one time job, I did via portal.
Create a new Storage Account: https://portal.azure.com/#blade/HubsExtension/BrowseResource/resourceType/Microsoft.Storage%2FStorageAccounts

In the newly created  storage account, add a new container with the desired name.

**Review by maly7 — APPROVED** [review]

**Pull request merged**